### PR TITLE
26 section testing types cdkinfrastructure testing

### LINKED ISSUE #26: Section: Testing types / Infrastructure testing
Points to write about:
- IaaC, CF, and CDK (maybe Terraform)
- Static testing (CF/CDK)
- Dynamic testing (Deployment validation) - Manually and Automated
- Example static/dynamic tests for [aws-static-site](https://github.com/idea-pool/aws-static-site)

Links
- https://docs.aws.amazon.com/cdk/v2/guide/testing.html
- https://aws.amazon.com/blogs/developer/testing-cdk-applications-in-any-language/
- https://github.com/cdklabs/aws-cdk-testing-examples
- https://cdkworkshop.com/20-typescript/70-advanced-topics/100-construct-testing.html
- https://kreuzwerker.de/post/infrastructure-tests-with-cdk
- https://thomasstep.com/blog/how-to-write-aws-cdk-tests
- https://aws.amazon.com/blogs/developer/testing-infrastructure-with-the-aws-cloud-development-kit-cdk/
- https://speakerdeck.com/bwhaley/best-practices-for-testing-aws-infrastructure
- https://pages.awscloud.com/Infrastructure-as-Code-Testing-Strategies-with-AWS-CloudFormation_2019_0504-DEV_OD.html

Source: https://github.com/Idea-Pool/aws-testing/blob/main/testing-types/cfk-infrastructure-testing.md

### PR DESCRIPTION
**Pull request by szikszail**

### CODE REVIEW

File: docs/testing-types/infrastructure-testing.md [hunk 1]
```diff
@@ -5,4 +5,295 @@ nav_order: 3
...
+Deployment validation is a core part of dynamic testing, which verifies that the infrastructure components are provisioned correctly and interact as expected.
+
+During deployment validation, you should test various aspects such as resource creation, modification, and deletion, as well as network connectivity, security configurations, and performance metrics.
+
+Dynamic testing can be performed manually by following predefined test scripts and conducting exploratory testing. To achieve higher efficiency and scalability, it is advisable to automate dynamic testing using custom scripts leveraging AWS SDKs.
+
+An example test suite for the example stack could be the following (using TypeScript, Jest, and AWS SDK):
+
+```typescript
+import AWS from 'aws-sdk';
```
Comment (kornel-filep) [inline docs/testing-types/infrastructure-testing.md]: I believe this is AWS SDK v2.
I think using AWS SDK V3 would be better (at least in the step function testing topic we are going to use v3) as it has a modular architecture with smaller packages.

Comment (szikszail) [reply docs/testing-types/infrastructure-testing.md]: Good point. I will update it.

File: docs/testing-types/infrastructure-testing.md [hunk 2]
```diff
@@ -5,4 +5,384 @@ nav_order: 3
...
+    "WebsiteUrl": {
+      "Description": "The URL of the CloudFront distribution",
+      "Value": {}
+    }
+  }
+}
+```
+
+{: .note }
+Note, that this is only a part of the synthetized JSON template and as it is now, not a valid and deployable CF JSON template.
```
Comment (kornel-filep) [inline docs/testing-types/infrastructure-testing.md]: I think clarifying what synthetized this template would make it more clear for the reader what is this.
```suggestion
Note, that this CF JSON template is synthetized by aws-cdk, and it is only a part of the synthetized JSON template and as it is now, not a valid and deployable CF JSON template by itself.
```

File: docs/testing-types/infrastructure-testing.md [hunk 3]
```diff
@@ -5,4 +5,384 @@ nav_order: 3
...
+    new cdk.CfnOutput(this, 'WebsiteUrl', {
+      value: `https://${distribution.domainName}`,
+      description: 'The URL of the CloudFront distribution'
+    });
+  }
+}
+
+```
+
+By synthesizing and deploying this AWS CDK code, AWS will provision the specified bucket and distributionwith the defined configurations.
```
Comment (kornel-filep) [inline docs/testing-types/infrastructure-testing.md]:
```suggestion
By synthesizing and deploying this AWS CDK code, AWS will provision the specified bucket and distribution with the defined configurations.
```

File: samples/static-site/README.md
Comment (kornel-filep) [inline samples/static-site/README.md:1]: Does this deployment require a proper awscli with credentials configured?
If yes, then adding a 1 sentence about this as a prerequisite would be nice I think

**Comment by kornel-filep** [comment]
For me the light/dark switcher appears as a block without any icons.
image

File: samples/static-site/README.md
Comment (szikszail) [reply samples/static-site/README.md:1]: No, it does not. However, I added one sentence about it as you suggested.

**Review by kornel-filep — APPROVED** [review]

**Pull request merged**